lib/inets/src/http_client/httpc.erl

```diff
@@ -53,11 +53,93 @@
 	 stop_service/1, 
 	 services/0, service_info/1]).
 
+
+% common types
+-export_type([request_id/0, profile/0, socket_opt/0]).
+
+% HTTP request types
+-export_type([ method/0, http_request/0
+             , header/0, content_type/0, status_code/0
+             , body/0, http_body_result/0, request_result/0]).
+
+% SSL options
+-export_type([ ssloption/0, http_option/0]).
+
+% HTTP Option types
+-export_type([ option_request/0, stream_to/0, body_format/0
+             , receiver/0, receiver_module/0, receiver_function/0, receiver_args/0
+             , cookie_header_opt/0]).
+
 -include_lib("inets/src/http_lib/http_internal.hrl").
 -include("httpc_internal.hrl").
 
 -define(DEFAULT_PROFILE, default).
 
+%%%=========================================================================
+%%%  Type specifications
+%%%=========================================================================
+
+-opaque request_id() :: reference().
```

Comment:
`request_id` looks strange in the documentation since it is a type without definition. 
I don't think it can be an `opaque` type since the purpose is that the user is matching `request_id`s for equality.
Compare with ssl_socket in the ssl module.
Probably the type should be `request_id() :: any()` or Dialyzers support for `opaque` or the definition of `opaque` must be revisited.

Reply:
Yes, good catch!